src/meta/meta_turtle.c

```diff
@@ -138,6 +138,38 @@ __metadata_load_bulk(WT_SESSION_IMPL *session)
     return (ret);
 }
 
+/*
+ * __turtle_validate_version --
+ *     Retrieve version numbers from the turtle file and validate them against our WiredTiger
+ *     version.
+ */
+static int
+__turtle_validate_version(WT_SESSION_IMPL *session)
+{
+    WT_DECL_RET;
+    uint32_t major, minor;
+    char *version_string;
+
+    WT_WITH_TURTLE_LOCK(
+      session, ret = __wt_turtle_read(session, WT_METADATA_VERSION, &version_string));
+
+    if (ret != 0)
+        WT_ERR_MSG(session, ret, "Unable to read version string from turtle file");
+
+    if ((ret = sscanf(version_string, "major=%u,minor=%u", &major, &minor)) != 2)
+        WT_ERR_MSG(session, ret, "Unable to read version string from turtle file");
```

Comment:
This string should be unique from the one above it. I suggest something like:
"Unable to parse turtle file version string"